src/ui/units/wizard/components/Dialogs/Settings/Settings.tsx

```diff
@@ -550,6 +557,38 @@ class DialogSettings extends React.PureComponent<InnerProps, State> {
         );
     }
 
+    renderWidgetSize() {
+        const {visualization} = this.props;
+        const isTableWidget = (
+            [WizardVisualizationId.FlatTable, WizardVisualizationId.PivotTable] as string[]
+        ).includes(visualization.id);
+
+        if (!isTableWidget || !Utils.isEnabledFeature(Feature.TableSize)) {
+            return null;
+        }
+
+        const sizes = Object.values(WidgetSize);
+        const selected = this.state.size ?? DEFAULT_WIDGET_SIZE;
+
+        return (
+            <div className={b('widget-size')}>
+                <span className={b('label')}>{i18n('wizard', 'label_widget-size')}</span>
+                <RadioButton
+                    value={selected}
+                    onChange={(event) => {
```

Comment:
`onUpdate` props accepts (value: string) => {} (instead of the event). Optional, but maybe it would look simpler